openmdao/core/group.py

```diff
@@ -1273,6 +1273,7 @@ def _solve_nonlinear(self):
         # TODO: It is pointless to run this ahead of non-iterative solvers.
         for sub in self.system_iter(recurse=True):
             if hasattr(sub, 'guess_nonlinear'):
+                self._transfer('nonlinear', 'fwd')
```

Comment:
Yeah, that's the only way around it, to assure that we scatter the minimum amount of data that we need.